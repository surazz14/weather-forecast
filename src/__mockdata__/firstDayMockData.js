const firstDayMockData = [
  {
    dt: 1595203200,
    main: {
      temp: 292.87,
      feels_like: 295.37,
      temp_min: 292.85,
      temp_max: 292.87,
      pressure: 1003,
      sea_level: 1003,
      grnd_level: 862,
      humidity: 94,
      temp_kf: 0.02,
    },
    weather: [
      {
        id: 501,
        main: "Rain",
        description: "moderate rain",
        icon: "10d",
      },
    ],
    clouds: {
      all: 100,
    },
    wind: {
      speed: 0.87,
      deg: 84,
    },
    visibility: 1343,
    pop: 1,
    rain: {
      "3h": 4.01,
    },
    sys: {
      pod: "d",
    },
    dt_txt: "2020-07-20 00:00:00",
  },
  {
    dt: 1595214000,
    main: {
      temp: 292.95,
      feels_like: 295.37,
      temp_min: 292.95,
      temp_max: 292.95,
      pressure: 1005,
      sea_level: 1005,
      grnd_level: 863,
      humidity: 94,
      temp_kf: 0,
    },
    weather: [
      {
        id: 501,
        main: "Rain",
        description: "moderate rain",
        icon: "10d",
      },
    ],
    clouds: {
      all: 100,
    },
    wind: {
      speed: 1.04,
      deg: 102,
    },
    visibility: 2124,
    pop: 1,
    rain: {
      "3h": 10.12,
    },
    sys: {
      pod: "d",
    },
    dt_txt: "2020-07-20 03:00:00",
  },
  {
    dt: 1595224800,
    main: {
      temp: 293.05,
      feels_like: 295.53,
      temp_min: 293.05,
      temp_max: 293.05,
      pressure: 1004,
      sea_level: 1004,
      grnd_level: 863,
      humidity: 94,
      temp_kf: 0,
    },
    weather: [
      {
        id: 502,
        main: "Rain",
        description: "heavy intensity rain",
        icon: "10d",
      },
    ],
    clouds: {
      all: 100,
    },
    wind: {
      speed: 1.01,
      deg: 89,
    },
    visibility: 3826,
    pop: 1,
    rain: {
      "3h": 15.14,
    },
    sys: {
      pod: "d",
    },
    dt_txt: "2020-07-20 06:00:00",
  },
  {
    dt: 1595235600,
    main: {
      temp: 293.57,
      feels_like: 296.59,
      temp_min: 293.57,
      temp_max: 293.57,
      pressure: 1003,
      sea_level: 1003,
      grnd_level: 862,
      humidity: 92,
      temp_kf: 0,
    },
    weather: [
      {
        id: 501,
        main: "Rain",
        description: "moderate rain",
        icon: "10d",
      },
    ],
    clouds: {
      all: 100,
    },
    wind: {
      speed: 0.35,
      deg: 92,
    },
    visibility: 10000,
    pop: 0.92,
    rain: {
      "3h": 5.83,
    },
    sys: {
      pod: "d",
    },
    dt_txt: "2020-07-20 09:00:00",
  },
  {
    dt: 1595246400,
    main: {
      temp: 293.79,
      feels_like: 295.87,
      temp_min: 293.79,
      temp_max: 293.79,
      pressure: 1003,
      sea_level: 1003,
      grnd_level: 862,
      humidity: 91,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: "Rain",
        description: "light rain",
        icon: "10d",
      },
    ],
    clouds: {
      all: 100,
    },
    wind: {
      speed: 1.72,
      deg: 130,
    },
    visibility: 10000,
    pop: 0.92,
    rain: {
      "3h": 0.43,
    },
    sys: {
      pod: "d",
    },
    dt_txt: "2020-07-20 12:00:00",
  },
  {
    dt: 1595257200,
    main: {
      temp: 293.07,
      feels_like: 295.14,
      temp_min: 293.07,
      temp_max: 293.07,
      pressure: 1005,
      sea_level: 1005,
      grnd_level: 864,
      humidity: 92,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: "Rain",
        description: "light rain",
        icon: "10n",
      },
    ],
    clouds: {
      all: 85,
    },
    wind: {
      speed: 1.39,
      deg: 129,
    },
    visibility: 10000,
    pop: 0.73,
    rain: {
      "3h": 0.42,
    },
    sys: {
      pod: "n",
    },
    dt_txt: "2020-07-20 15:00:00",
  },
  {
    dt: 1595268000,
    main: {
      temp: 292.88,
      feels_like: 294.88,
      temp_min: 292.88,
      temp_max: 292.88,
      pressure: 1007,
      sea_level: 1007,
      grnd_level: 865,
      humidity: 91,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: "Rain",
        description: "light rain",
        icon: "10n",
      },
    ],
    clouds: {
      all: 91,
    },
    wind: {
      speed: 1.27,
      deg: 105,
    },
    visibility: 10000,
    pop: 0.95,
    rain: {
      "3h": 1,
    },
    sys: {
      pod: "n",
    },
    dt_txt: "2020-07-20 18:00:00",
  },
  {
    dt: 1595278800,
    main: {
      temp: 292.57,
      feels_like: 294.84,
      temp_min: 292.57,
      temp_max: 292.57,
      pressure: 1006,
      sea_level: 1006,
      grnd_level: 864,
      humidity: 93,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: "Rain",
        description: "light rain",
        icon: "10n",
      },
    ],
    clouds: {
      all: 100,
    },
    wind: {
      speed: 0.91,
      deg: 95,
    },
    visibility: 10000,
    pop: 1,
    rain: {
      "3h": 2.36,
    },
    sys: {
      pod: "n",
    },
    dt_txt: "2020-07-20 21:00:00",
  },
];

export default firstDayMockData