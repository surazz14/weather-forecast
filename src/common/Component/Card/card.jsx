import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";

const useStyles = makeStyles({
  root: {
    maxWidth: 400,
    marginLeft: "3%",
  },
});

export default function InfoCard(props) {
  const classes = useStyles();

  return (
    <Card className={classes.root} onClick={props.onClick}>
      <CardContent>
        <Box display="flex" justifyContent="center">
          <Typography variant="h6" paragraph={true} component="h6">
            Date :{props.date}
            <br />
            Avg Temperture:{props.avgTemp}
            <br />
            Avg Visibility:{props.avgVisibililty}
          </Typography>
        </Box>
      </CardContent>
    </Card>
  );
}
