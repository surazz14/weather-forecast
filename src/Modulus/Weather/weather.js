import React, { useEffect, useState } from "react";
import { inject, observer } from "mobx-react";
import { Bar } from "react-chartjs-2";
import Box from "@material-ui/core/Box";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import Typography from "@material-ui/core/Typography";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
import Card from "../../common/Component/Card/card";
import Loading from "../../common/Component/loading/loading";
import Chart from "../../common/Component/Chart/chart";

function Weather(props) {
  const [value, setValue] = useState("fahrenheit");
  const [loading, setLoading] = useState(true);
  const [data, setData] = useState([]);

  const handleChange = (event) => {
    setValue(event.target.value);
  };

  const onClickCard = (data) => {
    setData(data);
  };

  const {
    weatherCollection: {
      getWeather,
      getFiveDaysWeather,
      getAverageTemperture,
      getAverageVisibility,
      toCelcius,
      toFahrenheit,
    },
  } = props;

  useEffect(() => {
    async function getWeatherData() {
      await getWeather();
      setLoading(false);
    }
    getWeatherData();
  }, [getWeather]);
  
  const responsive = {
    superLargeDesktop: {
      breakpoint: { max: 4000, min: 3000 },
      items: 3,
    },
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 3,
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 2,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
    },
  };
  
  if (loading) {
    return <Loading loading={loading} />;
  }
  return (
    <>
      <Box ml="15%" mr="15%" mt="2%">
        <Box display="flex" justifyContent="center">
          <RadioGroup
            aria-label="temperture"
            name="temperture"
            value={value}
            row={true}
            onChange={handleChange}
          >
            <FormControlLabel
              value="celcius"
              control={<Radio color="primary" />}
              label="Celcius"
            />
            <FormControlLabel
              value="fahrenheit"
              control={<Radio color="primary" />}
              label="Fahrenheit"
            />
          </RadioGroup>
        </Box>

        <Carousel responsive={responsive}>
          {getFiveDaysWeather()?.map((data, i) => (
            <Card
              key={i}
              date={data[0]?.dt_txt.slice(0, 10)}
              onClick={() => onClickCard(data)}
              avgTemp={
                value === "fahrenheit"
                  ? toFahrenheit(getAverageTemperture(data)) + "°F"
                  : toCelcius(getAverageTemperture(data)) + "°C"
              }
              avgVisibililty={getAverageVisibility(data)}
            />
          ))}
        </Carousel>
        <Box mt="5%">
          <Typography align="center" variant="h5" component="h2">
            {data.length > 0
              ? `Weather forecast Chart of ${data[0]?.dt_txt.slice(
                  0,
                  10
                )} in ${value}`
              : "Select Card to see daily weather forecast chart"}
          </Typography>
          <br />
          {data.length > 0 && (
            <Chart
              type={Bar}
              labels={data.map((data) => data.dt_txt.slice(11))}
              data={data.map((data) =>
                value === "fahrenheit"
                  ? toFahrenheit(data?.main?.temp)
                  : toCelcius(data?.main?.temp)
              )}
              height={30}
              width={80}
            />
          )}
        </Box>
      </Box>
    </>
  );
}

export default inject("weatherCollection")(observer(Weather));
