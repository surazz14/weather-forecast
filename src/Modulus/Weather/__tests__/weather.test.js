import React, { cloneElement } from "react";
import Weather from "../weather";
import { Provider } from "mobx-react";
import { shallow, mount } from "enzyme";
import Card from "../../../common/Component/Card/card";

import firstDayMockData from "../../../__mockdata__/firstDayMockData";
import store from "../../../entities/setUpStore";
import { Button, Typography, RadioGroup } from "@material-ui/core";

describe("WeatherCollection", () => {
  // it("should render correctly", () => {
  //   let wrapper = shallow(
  //     <Provider {...store}>
  //       <Weather />
  //     </Provider>
  //   );

  //   expect(wrapper.html()).toMatchSnapshot();
  // });

  describe("action", () => {
    it("close button click", () => {
      // const onClose = jest.fn();
      const handleSubmit = jest.fn();
      const setValue = jest.fn();
      const changeData = jest.fn();
      let wrapper = mount(
        <Provider {...store}>
          <Weather />
        </Provider>
      );
      const radio = wrapper.find('.test');
      const buttons = wrapper.find(Button);
      const p = wrapper.find(Typography);
      const inputs = wrapper.find('.in')

      // console.log(radio.debug(),"this is debug")
      console.log(
        "this is value"
      );
      inputs.simulate("change")

      // expect(radio.props('value')).toBe('celcius')
      // buttons.simulate("click");
      // wrapper.update()
      console.log(p.at(2).debug());
      // expect(p.at(2).props().children).toBe('celcius')
      // console.log(radio.props().value,'radio')
      // console.log(p.at(2).props().children,"typo")
      const news = wrapper.find(Typography);
      // console.log(news.at(2).debug(), "this is new");

      // wrapper.update()
      // const handleChange = jest.spyOn(React, "useState");        // console.log(radio.props().value,'radio')
      // console.log(p.at(2).props().children,"typo")
      // handleChange.mockImplementation(value => [value, setValue])
      // expect(changeData).toHaveBeenCalled()
      expect(p.at(2).text()).toBe("celcius");
    });
  });
});
