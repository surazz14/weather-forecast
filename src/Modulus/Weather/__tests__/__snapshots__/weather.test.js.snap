// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCollection should create entity 1`] = `"<div class=\\"MuiBackdrop-root makeStyles-backdrop-1\\" aria-hidden=\\"true\\" style=\\"opacity:0\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-indeterminate\\" style=\\"width:40px;height:40px\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div>"`;

exports[`WeatherCollection should render correctly 1`] = `"<div class=\\"MuiBackdrop-root makeStyles-backdrop-1\\" aria-hidden=\\"true\\" style=\\"opacity:0\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-indeterminate\\" style=\\"width:40px;height:40px\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div>"`;
